README.md

```diff
@@ -1,84 +1,100 @@
-`TEST_TMPDIR` convention instead, so that I could test code that works with
-the file system.
+The search index is generated automatically via `jekyll build` or `jekyll
+serve`. The supporting JavaScript code is optimized and loads asynchronously.
+These features ensure that the preparation of the search index does not
+introduce rendering latency in the browser.
 
 ### Installation
 
-Add this line to your application's Gemfile:
+Add this line to your Jekyll project's `Gemfile`:
 
 ```ruby
-gem 'test_temp_file_helper'
+group :jekyll_plugins do
+  gem 'jekyll_pages_api_search'
+end
 ```
 
-And then execute:
-```
-$ bundle
+Then add the following to the project's `_config.yml` file:
+
+```yaml
+# Configuration for jekyll_pages_api_search plugin gem.
+jekyll_pages_api_search:
+  # Uncomment this to speed up site generation while developing.
+  #skip_index: true
+
+  # Each member of `index_fields` should correspond to a field generated by
+  # the jekyll_pages_api. It can hold an optional `boost` member as a signal
+  # to Lunr.js to weight the field more highly (default is 1).
+  index_fields:
+    title:
+      boost: 10
+    tags:
+      boost: 10
+    url:
+      boost: 5
+    body:
 ```
 
-Or install it yourself as:
-```
-$ gem install test_temp_file_helper
-```
+Now running `jekyll build` or `jekyll serve` will produce `search-index.json`
+and `search-index.json.gz` files in the `_site` directory (or other output
+directory, as configured).
 
 ### Usage
 
-Create a `TempFileHelper` in your test's `setup` method, and call
-`TempFileHelper.teardown` in your test's `teardown` method.
+To add the index to your pages, insert the following tags in your `_layouts`
+and `_includes` files as you see fit:
 
-```ruby
-require 'minitest/autorun'
-require 'test_temp_file_helper'
-
-class MyTest < ::Minitest::Test
-  def setup
-    @temp_file_helper = TestTempFileHelper::TempFileHelper.new
-  end
-
-  def teardown
-    @temp_file_helper.teardown
-  end
-
-  def test_something_that_handles_files
-    dir_path = @temp_file_helper.mkdir(File.join('path', 'to', 'dir'))
-    file_path = @temp_file_helper.mkfile(File.join('path', 'to', 'file'))
-    ...
-  end
-end
-```
+- `{% jekyll_pages_api_search_interface %}`: inserts the HTML for the search
+  box and search results
+- `{% jekyll_pages_api_search_load %}`: inserts the `<script>` tags to load
+  the search code asynchronously; the search code will then load
+  `search-index.json` asynchronously
+
+You can also add `@import "jekyll_pages_api_search";` to one of your [Sass
+assets](http://jekyllrb.com/docs/assets/) to use the default interface style.
+
+### Customization
+
+If you prefer to craft your own versions of these tags and styles, you can
+capture the output of these tags and the Sass `@import` statement, then create
+new tags or included files based on this output, careful not to change
+anything that causes the interaction between these components to fail.
```

Comment:
Not sure that this is enough information for someone to go on...I might consider just leaving it out until we have a better idea of what customizations people would want.